doc/problem/behavior.md

```diff
@@ -52,3 +59,40 @@ For In/Active State:
 - AS Probability: Already in basic time bin features.
 - AS Duration: Already in basic time bin features.
 - AS Rate: Use AS Number over AS Duration.
+
+
+## Data Requirements Description
+The data we have:
+- a dataframe of observations of location for each mice, (x, y, t) with \Delta t small.
+
+The data we require:
+- The above dataframe with AS/ IS properly classified at each point t for each mouse
+- The above dataframe with a classification of drinking, feeding and locomotion for each mouse at each time t
+- The above dataframe with a classification of consumption size and moving distance with each event at each time t
+- The above dataframe with a classification of strain number and mouse number at each time t (if available)
+
+## Methodology/ Approach Description
+We wish to create a single function that should be able to return all of the above metrics as a list:
+- Key inputs are:
+    - mouse/ strain as string
+    - starting time
+    - ending time
+    - a dictionary containing the rectangular vertices marking the area to restrict the movement to i.e. x_lower, x_upper, y_lower, y_upper.
+    - [_Drinking | Feeding | Locomotion_] state specification
+- The main output is a list containing the key metrics stated in `Statement of statistical problems` section
+
+- Key idea is that if we have the most granular dataframe in `Data Requirements Description` then the python code is really just a SQL (in `pandas` form) filtering/ grouping query to generate the required output metrics (from flowchart) in the form of a list
+
+## Testing Framework Outline
+
+## Additional Remarks
+- It is not clear exactly how the existing metrics are to be calculated in the form a single query
+- Not sure yet whether the required dataframe at the most granular level can be easily constructed. This would be really useful for all projects to use so we should really consider developing it for the wider team.
+- Some of the required data metrics like consumption of food/ water at each time t may not be easy to obtain as they are provided for each interval. These may have to be prorated across each time t in some stable way in the construction of the required dataframe
+- We also believe that the metrics provided at each point are single point statistics i.e. means. We should consider outputing the actual histogram of values at each point for the given metric rather than just the single-valued mean metrics
```

Comment:
For example, we may not only be interested in the average amount of active time spent in locomotion, but the distribution of locomotion. This is a more complicated metric than those outlined in the work by the Tecott Lab's papers referenced below. With this information, we could potentially see interesting trends: the proportion of a mouse-day spent in locomotion could be the same in two time chunks, but the types of movements (distances) could form a more nuanced distribution.